projects/components/src/input-pill-list/input-pill-list.component.ts

```diff
@@ -0,0 +1,183 @@
+import { ChangeDetectionStrategy, Component, EventEmitter, Input, OnChanges, Output } from '@angular/core';
+import {
+  ControlValueAccessor,
+  FormArray,
+  FormBuilder,
+  FormControl,
+  FormGroup,
+  NG_VALUE_ACCESSOR,
+  Validators
+} from '@angular/forms';
+import { IconType } from '@hypertrace/assets-library';
+import { isEnterOrCommaKeyEvent, TypedSimpleChanges } from '@hypertrace/common';
+import { debounce, isEmpty } from 'lodash-es';
+import { IconSize } from '../icon/icon-size';
+import { InputAppearance } from '../input/input-appearance';
+
+@Component({
+  selector: 'ht-input-pill-list',
+  template: `
+    <div class="input-pill-list" [formGroup]="this.form">
+      <div class="header">
+        <ht-form-field class="form-field">
+          <ht-input
+            [disabled]="this.shouldDisableUpdate || this.disableAdd"
+            class="input primary-input"
+            formControlName="inputBuffer"
+            (keydown)="this.addBufferValueToList($event)"
+            placeholder="Enter a comma separated list of values"
+            appearance="${InputAppearance.Border}"
+          ></ht-input>
+        </ht-form-field>
+      </div>
+
+      <ng-container formArrayName="pillControls">
+        <div class="pill-list" *ngIf="this.pillControlsArray.value.length > 0">
+          <div class="pill" *ngFor="let pillValueControl of this.pillControlsArray?.controls; index as index">
+            <ht-form-field class="form-field" [formGroup]="pillValueControl">
+              <ht-input
+                class="input secondary-input"
+                [disabled]="this.shouldDisableUpdate"
+                appearance="${InputAppearance.Border}"
+                (valueChange)="this.updateValue()"
+                formControlName="value"
+              ></ht-input>
+            </ht-form-field>
+            <ht-icon
+              class="close-icon"
+              [ngClass]="{ disabled: this.shouldDisableUpdate }"
+              icon="${IconType.CloseCircle}"
+              size="${IconSize.Small}"
+              (click)="this.removeValue(index)"
+            ></ht-icon>
+          </div>
+        </div>
+      </ng-container>
+    </div>
+  `,
+  styleUrls: ['./input-pill-list.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      multi: true,
+      useExisting: InputPillListComponent
+    }
+  ]
+})
+export class InputPillListComponent implements ControlValueAccessor, OnChanges {
+  @Input()
+  public readonly values: string[] = [];
+
+  @Input()
+  public readonly disableAdd: boolean = false;
+
+  @Input()
+  public readonly disableUpdate: boolean = false;
```

Comment:
is it only used for removing the pill?

Reply:
also to control editing the values of pills.